bin/lock.lua

```diff
@@ -0,0 +1,141 @@
+os.pullEvent = os.pullEventRaw
+
+local lock = {
+    '  888  ',
+    ' 8   8 ',
+    ' 8   8 ',
+    '8888888',
+    '88fff88',
+    '888f888',
+    '888f888',
+    ' 88888 ',
+}
+local lockWidth = #lock[1]
+local lockHeight = #lock
+local lockString = ''
+for _, v in ipairs(lock) do
+    lockString = lockString .. v .. '\n'
+end
+local lockImage = paintutils.parseImage(lockString)
+
+local unlock = {
+    '  888  ',
+    ' 8   8 ',
+    ' 8   8 ',
+    ' 8     ',
+    '8888888',
+    '88fff88',
+    '888f888',
+    '888f888',
+    ' 88888 ',
+}
+local unlockString = ''
+for _, v in ipairs(unlock) do
+    unlockString = unlockString .. v .. '\n'
+end
+local unlockImage = paintutils.parseImage(unlockString)
+
+term.clear()
+
+local code = ...
+code = tonumber(code)
+
+print('Please place door or redstone to the right of this computer')
+sleep(2)
+
+redstone.setOutput('right', true)
+
+if not code or #tostring(code) ~= 5 then
+    print('Please specify the code to use (5 Numbers)')
+    return;
+end
+
+local monitor
+
+for _, v in pairs({peripheral.find('monitor')}) do
+    local width, height = v.getSize()
+
+    if width == 7 and height == 5 then
+        monitor = v
+    end
+end
+
+if not monitor then
+    print('Please connect a 1x1 advanced monitor')
+    return;
+end
```

Comment:
is it possible to support other resolutions with e.g. scaling? also, what if multiple monitors are connected?

Reply:
because of how I coded it, not really. That is a good question, and I am ignoring it